app/src/marketplace/components/NewProductPage.jsx

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import React, { useEffect, useState } from 'react'
+import { type Location } from 'react-router-dom'
+import styled from 'styled-components'
+import qs from 'query-string'
+import { useDispatch } from 'react-redux'
+import { replace } from 'connected-react-router'
+import { postEmptyProduct } from '$mp/modules/deprecated/editProduct/services'
+import LoadingIndicator from '$userpages/components/LoadingIndicator'
+import routes from '$routes'
+import { type ProductType } from '$mp/flowtype/product-types'
+import useIsMounted from '$shared/hooks/useIsMounted'
+import { productTypes } from '$mp/utils/constants'
+
+type Props = {
+    className?: ?string,
+    location: Location,
+}
+
+const sanitizedType = (type: ?string): ProductType => (
+    productTypes[(type || '').toUpperCase()] || productTypes.NORMAL
+)
+
+const UnstyledNewProductPage = ({ className, location: { search } }: Props) => {
+    const dispatch = useDispatch()
+
+    const isMounted = useIsMounted()
+
+    const [error, setError] = useState()
+
+    useEffect(() => {
+        const { type } = qs.parse(search)
+
+        postEmptyProduct(sanitizedType(type))
+            .then(({ id }) => {
+                if (isMounted()) {
+                    dispatch(replace(routes.editProduct({
+                        id,
+                    })))
+                }
+            })
+            .catch(setError)
```

Comment:
Hm, this also catches errors in the `then` e.g. the `dispatch` reducer. Seems unlikely to happen but not sure such errors should be handled by this `.catch`. Perhaps use second argument to `.then`? I guess this is just ported from the existing code.

Reply:
Done in e0fbc69.